src/coordinator/storage/local/config.go

```diff
@@ -0,0 +1,194 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package local
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/m3db/m3db/src/coordinator/storage"
+	"github.com/m3db/m3db/src/dbnode/client"
+	"github.com/m3db/m3x/ident"
+)
+
+var (
+	defaultNewClientConfigurationParams = client.ConfigurationParameters{}
+)
+
+// ClustersStaticConfiguration is a set of static cluster configurations.
+type ClustersStaticConfiguration []ClusterStaticConfiguration
+
+type newClientFromConfig func(
+	cfg client.Configuration,
+	params client.ConfigurationParameters,
+	custom ...client.CustomOption,
+) (client.Client, error)
+
+// ClusterStaticConfiguration is a static cluster configuration.
+type ClusterStaticConfiguration struct {
+	newClientFromConfig newClientFromConfig
+	Client              client.Configuration                  `yaml:"client"`
+	Namespaces          []ClusterStaticNamespaceConfiguration `yaml:"namespaces"`
+}
+
+func (c ClusterStaticConfiguration) newClient(
+	params client.ConfigurationParameters,
+	custom ...client.CustomOption,
+) (client.Client, error) {
+	if c.newClientFromConfig != nil {
+		return c.newClientFromConfig(c.Client, params, custom...)
+	}
+	return c.Client.NewClient(params, custom...)
+}
+
+// ClusterStaticNamespaceConfiguration describes the namespaces in a
+// static cluster.
+type ClusterStaticNamespaceConfiguration struct {
+	Namespace          string              `yaml:"namespace"`
+	StorageMetricsType storage.MetricsType `yaml:"storageMetricsType"`
+	Retention          time.Duration       `yaml:"retention" validate:"nonzero"`
+	Resolution         time.Duration       `yaml:"resolution" validate:"min=0"`
+}
+
+type unaggregatedClusterNamespaceConfiguration struct {
+	client    client.Client
+	namespace ClusterStaticNamespaceConfiguration
+	result    clusterConnectResult
+}
+
+type aggregatedClusterNamespacesConfiguration struct {
+	client     client.Client
+	namespaces []ClusterStaticNamespaceConfiguration
+	result     clusterConnectResult
+}
+
+type clusterConnectResult struct {
+	session client.Session
+	err     error
+}
+
+// NewClusters instantiates a new Clusters instance.
+func (c ClustersStaticConfiguration) NewClusters() (Clusters, error) {
+	var (
+		numUnaggregatedClusterNamespaces int
+		numAggregatedClusterNamespaces   int
+		unaggregatedClusterNamespaceCfg  = &unaggregatedClusterNamespaceConfiguration{}
+		aggregatedClusterNamespacesCfgs  []*aggregatedClusterNamespacesConfiguration
```

Comment:
So the main reason I'm using pointers here is that I need to set the client session and error back on the struct itself once I try connecting to the cluster.  It seemed simpler just to use a pointer, and since this is a slow path (i.e. once at initialization) I didn't think too much about it.